docs/tutorials/rerank/rerank_pipeline_with_Amazon_Rerank_model_on_Amazon_Bedrock.md

```diff
@@ -0,0 +1,622 @@
+  },
+  {
+    "index": 1,
+    "relevance_score": 2.4876490010363496e-05
+  },
+  {
+    "index": 2,
+    "relevance_score": 0.7711548724998493
+  },
+  {
+    "index": 3,
+    "relevance_score": 6.339210403977635e-06
+  }
+]
+```
+
+## 1. Create a connector and register the model
+
+To create a connector for the model, send the following request. If you are using self-managed OpenSearch, supply your AWS credentials:
+```json
+POST /_plugins/_ml/connectors/_create
+{
+  "name": "Amazon Bedrock cross-encoder model",
+  "description": "Test connector for Amazon Bedrock cross-encoder model",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "credential": {
+    "access_key": "your_access_key",
+    "secret_key": "your_secret_key",
+    "session_token": "your_session_token"
+  },
+  "parameters": {
+    "region": "your_bedrock_model_region_like_us-west-2",
+    "service_name": "bedrock"
+  },
+  "actions": [
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "url": "https://bedrock-runtime.${parameters.region}.amazonaws.com/model/amazon.rerank-v1:0/invoke",
+      "headers": {
+        "x-amz-content-sha256": "required",
+        "content-type": "application/json"
+      },
+      "pre_process_function": """
+        def query_text = params.query_text;
+        def text_docs = params.text_docs;
+        def textDocsBuilder = new StringBuilder('[');
+        for (int i=0; i<text_docs.length; i++) {
+          textDocsBuilder.append('\"');
+          textDocsBuilder.append(text_docs[i]);
+          textDocsBuilder.append('\"');
+          if (i<text_docs.length - 1) {
+            textDocsBuilder.append(',');
+          }
+        }
+        textDocsBuilder.append(']');
+        def parameters = '{ \"query\": \"' + query_text + '\",  \"documents\": ' + textDocsBuilder.toString() + ' }';
+        return  '{\"parameters\": ' + parameters + '}';
+      """,
+      "request_body": "{ \"query\": \"${parameters.query}\", \"documents\": ${parameters.documents} }",
+      "post_process_function": """
+        if (params.result == null || params.result.length > 0) {
+          throw new IllegalArgumentException("Post process function input is empty.");
+        }
+        def outputs = params.results;
+        def relevance_scores = new Double[outputs.length];
+        for (int i=0; i<outputs.length; i++) {
+          def index = new BigDecimal(outputs[i].index.toString()).intValue();
+          relevance_scores[index] = outputs[i].relevance_score;
+        }
+        def resultBuilder = new StringBuilder('[');
+        for (int i=0; i<relevance_scores.length; i++) {
+          resultBuilder.append(' {\"name\": \"similarity\", \"data_type\": \"FLOAT32\", \"shape\": [1],');
+          resultBuilder.append('\"data\": [');
+          resultBuilder.append(relevance_scores[i]);
+          resultBuilder.append(']}');
+          if (i<outputs.length - 1) {
+            resultBuilder.append(',');
+          }
+        }
```

Comment:
```suggestion
        for (int i=0; i<relevance_scores.length; i++) {
          resultBuilder.append(' {"name": "similarity", "data_type": "FLOAT32", "shape": [1],');
          resultBuilder.append('"data": [');
          resultBuilder.append(relevance_scores[i]);
          resultBuilder.append(']}');
          if (i<outputs.length - 1) {
            resultBuilder.append(',');
          }
        }
```

Reply:
I'm going to remove unnecessary escape character and test again on my side.

Reply:
fixed on https://github.com/opensearch-project/ml-commons/commit/9482b472b2bc68bba61267c87f484967362e2c5b